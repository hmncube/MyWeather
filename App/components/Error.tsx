import React from 'react';
import {View, StyleSheet, Text} from 'react-native';
import darkTheme from '../constants/darkTheme';
import lightTheme from '../constants/lightTheme';
import Button from './Button';
import RNRestart from 'react-native-restart';

const Error = ({msg, isDarkMode}) => {
  return (
    <View style={styles(isDarkMode).container}>
      <View style={styles(isDarkMode).errorContainer}>
        <Text style={[styles(isDarkMode).text, styles(isDarkMode).textTitle]}>
          Sorry something went wrong
        </Text>
        <Text style={[styles(isDarkMode).text, styles(isDarkMode).textTech]}>
          Technical Error message
        </Text>
        <Text style={[styles(isDarkMode).text, styles(isDarkMode).textTechMsg]}>
          "{msg}"
        </Text>
        <Button
          onPress={() => RNRestart.Restart()}
          title={"Let's start over"}
          isDarkMode={isDarkMode}
        />
      </View>
    </View>
  );
};

const styles = isDarkMode =>
  StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: isDarkMode
        ? darkTheme.background
        : lightTheme.background,
    },
    errorContainer: {
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: isDarkMode ? darkTheme.error : lightTheme.error,
      padding: 60,
      borderWidth: 1,
      borderRadius: 11,
      borderColor: isDarkMode ? darkTheme.surface : lightTheme.surface,
    },
    text: {
      color: isDarkMode ? darkTheme.onError : lightTheme.onError,
      textAlign: 'center',
    },
    textTitle: {
      fontWeight: 'bold',
      fontSize: 22,
    },
    textTech: {
      marginTop: 4,
      fontSize: 18,
    },
    textTechMsg: {
      marginTop: 4,
      fontSize: 16,
      fontStyle: 'italic',
    },
  });

export default Error;
