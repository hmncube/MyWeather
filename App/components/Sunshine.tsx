import React from 'react';
import {Text, View, StyleSheet} from 'react-native';
import Card from '../components/Card';
import darkTheme from '../constants/darkTheme';
import lightTheme from '../constants/lightTheme';

const Sunshine = ({sunrise, sunset, length, remaining, isDarkMode}) => {
  const view = createView(sunrise, sunset, length, remaining, isDarkMode);
  return <Card view={view} isDarkMode={isDarkMode} />;
};

const createView = (sunrise, sunset, length, remaining, isDarkMode) => {
  return (
    <View>
      {createColumnView('SUNRISE', sunrise, isDarkMode)}
      {createColumnView('SUNSET', sunset, isDarkMode)}
      {createColumnView('LENGTH OF DAY', length, isDarkMode)}
      {createColumnView('REMAINING DAYLIGHT', remaining, isDarkMode)}
    </View>
  );
};
const createColumnView = (title, value, isDarkMode) => {
  return (
    <View style={styles(isDarkMode).columnItem}>
      <Text style={styles(isDarkMode).textTitle}>{title}</Text>
      <Text style={styles(isDarkMode).textValue}>{value}</Text>
    </View>
  );
};
const styles = isDarkMode =>
  StyleSheet.create({
    textTitle: {
      color: isDarkMode ? darkTheme.onBackground : lightTheme.gray77,
      fontSize: 12,
      textAlign: 'center',
      marginLeft: 16,
    },
    textValue: {
      flex: 1,
      color: isDarkMode ? darkTheme.onBackground : lightTheme.onSurface,
      fontSize: 15,
      textAlign: 'right',
      marginRight: 16,
    },
    row: {
      flexDirection: 'row',
      alignSelf: 'stretch',
      padding: 8,
      backgroundColor: isDarkMode ? darkTheme.surface : lightTheme.surface,
    },
    columnItem: {
      flexDirection: 'row',
      alignSelf: 'stretch',
      padding: 8,
    },
  });

export default Sunshine;
