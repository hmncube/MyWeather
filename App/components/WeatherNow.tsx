import React from 'react';
import {Text, View, StyleSheet} from 'react-native';
import Card from '../components/Card';
import darkTheme from '../constants/darkTheme';
import lightTheme from '../constants/lightTheme';

const WeatherNow = ({description, pressure, humidity, isDarkMode}) => {
  const view = createView(description, pressure, humidity, isDarkMode);
  return <Card view={view} isDarkMode={isDarkMode} />;
};

const createView = (description, pressure, humidity, isDarkMode) => {
  return (
    <View style={styles(isDarkMode).row}>
      {createRowView('DESCRIPTION', description, isDarkMode)}
      {createRowView('PRESSURE', pressure, isDarkMode)}
      {createRowView('HUMIDITY', humidity, isDarkMode)}
    </View>
  );
};

const createRowView = (title, value, isDarkMode) => {
  return (
    <View style={styles(isDarkMode).rowItem}>
      <View>
        <Text style={styles(isDarkMode).textTitle}>{title}</Text>
        <Text style={styles(isDarkMode).textValue}>{value}</Text>
      </View>
    </View>
  );
};

const styles = isDarkMode =>
  StyleSheet.create({
    textTitle: {
      color: isDarkMode ? darkTheme.onBackground : lightTheme.gray77,
      fontSize: 12,
      textAlign: 'center',
    },
    textValue: {
      color: isDarkMode ? darkTheme.onBackground : lightTheme.darkGray,
      fontSize: 15,
      textAlign: 'center',
    },
    row: {
      flexDirection: 'row',
      alignSelf: 'stretch',
      padding: 8,
      backgroundColor: isDarkMode ? darkTheme.surface : lightTheme.surface,
    },
    rowItem: {
      flex: 1,
    },
  });

export default WeatherNow;
