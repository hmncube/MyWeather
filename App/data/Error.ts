export class Error {
  isError: Boolean;
  msg: string;

  constructor(isError: Boolean, msg: string) {
    this.isError = isError;
    this.msg = msg;
  }
}
