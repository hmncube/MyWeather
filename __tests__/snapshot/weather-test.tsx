import React from 'react';
import renderer from 'react-test-renderer';
import Sunshine from '../../App/components/Sunshine';
import WeatherNow from '../../App/components/WeatherNow';
import Error from '../../App/components/Error';
//import Loading from '../../App/components/Loading';

describe('Testing all components for the weather screen', () => {
  it('Sunshine renders correctly', () => {
    const tree = renderer
      .create(
        <Sunshine
          sunrise={'0700AM'}
          sunset={'0700PM'}
          length={'12 Hrs 0 min'}
          remaining={'10 Hrs 2 min'}
          isDarkMode={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('WeatherNow renders correctly', () => {
  const tree = renderer
    .create(
      <WeatherNow
        description={'CLOUDY'}
        pressure={'1000 hPa'}
        humidity={'30%'}
        isDarkMode={true}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Error renders correctly', () => {
  const tree = renderer
    .create(<Error msg={'Test erroor message'} isDarkMode={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//TODO fix test
// it('Loading renders correctly', () => {
//   const tree = renderer.create(<Loading />).toJSON();
//   expect(tree).toMatchSnapshot();
// });
