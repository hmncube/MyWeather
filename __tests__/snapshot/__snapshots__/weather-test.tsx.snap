// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#121212",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#CF6679",
        "borderColor": "#262626",
        "borderRadius": 11,
        "borderWidth": 1,
        "justifyContent": "center",
        "padding": 60,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "textAlign": "center",
          },
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sorry something went wrong
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "textAlign": "center",
          },
          Object {
            "fontSize": 18,
            "marginTop": 4,
          },
        ]
      }
    >
      Technical Error message
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "textAlign": "center",
          },
          Object {
            "fontSize": 16,
            "fontStyle": "italic",
            "marginTop": 4,
          },
        ]
      }
    >
      "
      Test erroor message
      "
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#BB86FC",
          "borderRadius": 4,
          "elevation": 3,
          "justifyContent": "center",
          "marginHorizontal": 12,
          "marginTop": 16,
          "marginVertical": 6,
          "paddingHorizontal": 32,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Let's start over
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing all components for the weather screen Sunshine renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#262626",
      "borderColor": "#262626",
      "borderRadius": 11,
      "borderWidth": 1,
      "margin": 8,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flexDirection": "row",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 12,
            "marginLeft": 16,
            "textAlign": "center",
          }
        }
      >
        SUNRISE
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontSize": 15,
            "marginRight": 16,
            "textAlign": "right",
          }
        }
      >
        0700AM
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flexDirection": "row",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 12,
            "marginLeft": 16,
            "textAlign": "center",
          }
        }
      >
        SUNSET
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontSize": 15,
            "marginRight": 16,
            "textAlign": "right",
          }
        }
      >
        0700PM
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flexDirection": "row",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 12,
            "marginLeft": 16,
            "textAlign": "center",
          }
        }
      >
        LENGTH OF DAY
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontSize": 15,
            "marginRight": 16,
            "textAlign": "right",
          }
        }
      >
        12 Hrs 0 min
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flexDirection": "row",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 12,
            "marginLeft": 16,
            "textAlign": "center",
          }
        }
      >
        REMAINING DAYLIGHT
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontSize": 15,
            "marginRight": 16,
            "textAlign": "right",
          }
        }
      >
        10 Hrs 2 min
      </Text>
    </View>
  </View>
</View>
`;

exports[`WeatherNow renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#262626",
      "borderColor": "#262626",
      "borderRadius": 11,
      "borderWidth": 1,
      "margin": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#262626",
        "flexDirection": "row",
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          DESCRIPTION
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          CLOUDY
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          PRESSURE
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          1000 hPa
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          HUMIDITY
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          30%
        </Text>
      </View>
    </View>
  </View>
</View>
`;
